src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketBuildStatus.java

```diff
@@ -107,6 +108,15 @@ public String getUrl() {
         return url;
     }
 
+    public String getSignature() {
```

Comment:
I think we should not build a `string` for this, just use `signature.update` and add in the fields in the right order. That way you should be able to remove this method altogether.